MAINT-52379:Password can be change to the same one

Noussairhamrit opened the pull request:
ISSUE : Password can be change to the same one
FIX : add another condition that will process the error code sent by the back-end part and if the case it will trigger an error message for that I also add a constant UNCHANGED_NEW_PASSWORD_ERROR_CODE and also adding a new error message into UserSettings_en .properties

houssemexo26 committed:
TASK-35704: Create FB maintenance and update projects versions/dependencies

Amine0008 committed:
49308: Add Ellipsis for the blue chips on the settings page for responsive layout (#1129)

ISSUE: in the settings page we need to display the blue label chips with ellipsis for mobile web
FIX: added custom style that will be applied to these chips only when the application is opened on a mobile browser

Amine0008 committed:
changed wraper element for the v-chips components to conserve their default behaviour of resizing (#1146)

ISSUE: the previous method of applying ellipses include hard-coded width length for the label chips
FIX: changed the wrapper element for the v-chip components to preserve their default resizing behavior
Co-authored-by: Ali HAMDI <ahamdi@exoplatform.com>

Amine0008 committed:
task#37195: Mobile profile page issue with call icon (#1153)

ISSUE: margin ma-2 causes misalignment for mobile browsers
FIX: changed margin to ma-1 for mobile browsers

Amine0008 committed:
task#47237: remove avatar inside comment block (#1164)

ISSUE: UX improvement, increasing the comment block size by removing the avatar image
FIX: removed the avatar image for comments and readjusted the margin styling

Amine0008 committed:
undid changes in task47237 where we removed the users avatar for the activity  stream page (#1167)

ISSUE: reverting to previous behaviour where we removed the users avatar
FIX: undid changes in task 47237

hakermi committed:
MAINT-52378: Remove unneeded vuetify password lenght validators in reset user password settings (#1170)

ISSUE: Unneded pasword validators used in password validation while there are already validators for lenght and format in backend
FIX: Remove the unneded validators

Noussairhamrit committed:
MAINT-52379:Password can be change to the same one

ghost commented:
### CodeSee Review Map:
[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/Meeds-io/social/1171/21160a71/b8a3bee6186cc7c258c392083bb84dabecb5f328.svg)](https://app.codesee.io/r/reviews?pr=1171&src=https%3A%2F%2Fgithub.com%2FMeeds-io%2Fsocial)

Review [in an interactive map](https://app.codesee.io/r/reviews?pr=1171&src=https%3A%2F%2Fgithub.com%2FMeeds-io%2Fsocial)

View [more CodeSee Maps](https://app.codesee.io/r/maps?src=https%3A%2F%2Fgithub.com%2FMeeds-io%2Fsocial)

#### Legend
![CodeSee Map Legend](https://app.codesee.io/Legend_v2.png)

rdenarie commented on webapp/portlet/src/main/webapp/vue-apps/user-setting-security/components/UserSettingSecurityWindow.vue:
```diff
@@ -144,13 +144,20 @@ export default {
         }
       }
 
+
```
Please remove this modificaiton

rdenarie commented on webapp/portlet/src/main/webapp/vue-apps/user-setting-security/components/UserSettingSecurityWindow.vue:
```diff
@@ -144,13 +144,20 @@ export default {
         return;
       }
 
+      if (this.newPassword === this.currentPassword) {
+        this.$refs.confirmNewPassword.setCustomValidity(this.$t('UserSettings.label.changePasswordIdentical'));
+        if (!this.$refs.form.$el.reportValidity()) {
+          return;
+        }
+      }
```
As explained in the task, this verification must be done on server side. It will be use later to check more than one password in password history.
Please, can you redo this PR by making this check on server side

IlhemEssaadi committed:
TASK-52377:fixed Confirm password filled in user administration (#1169)

ISSUES : the field confirm password is already filled when adding a new user.
FIX : change v-model to take as value user.confirmNewPassword instead of confirmNewPassword because all the modifications are carried out on the user variable as well as the reset when saving a new user.

abenmbarek committed:
TASK-51126 : User's profile: Spaces in common: incorrect values (#1155)

ISSUES : incorrect counting of common spaces values in the rest service
FIX : add a new SQL query that returns the common spaces between two given users and use it into a new added service.
without adding unit tests

Noussairhamrit committed:
clean up code

Noussairhamrit commented:
related PR:
https://github.com/Meeds-io/gatein-portal/pull/306

rdenarie commented on webapp/portlet/src/main/webapp/vue-apps/user-setting-security/components/UserSettingSecurityWindow.vue:
```diff
@@ -106,6 +106,7 @@
 const USER_NOT_FOUND_ERROR_CODE = 'USER_NOT_FOUND';
 const WRONG_USER_PASSWORD_ERROR_CODE = 'WRONG_USER_PASSWORD';
 const PASSWORD_UNKNOWN_ERROR_CODE = 'PASSWORD_UNKNOWN_ERROR_CODE';
+const WRONG_NEW_PASSWORD_ERROR_CODE = 'WRONG_NEW_PASSWORD';
```
Update this line according to modifications asked in the other PR

rdenarie commented on webapp/portlet/src/main/webapp/vue-apps/user-setting-security/components/UserSettingSecurityWindow.vue:
```diff
@@ -143,14 +144,12 @@ export default {
       if (!this.$refs.form.$el.reportValidity()) {
         return;
       }
-
```
Remove this modification

rdenarie commented on webapp/portlet/src/main/webapp/vue-apps/user-setting-security/components/UserSettingSecurityWindow.vue:
```diff
@@ -143,14 +144,12 @@ export default {
           return;
         }
       }
-
```
Remove this modification

Noussairhamrit committed:
code refactoring

rdenarie committed:
Merge branch 'feature/maintenance' into fix/52379

rdenarie reviewed (changes requested):
2 feedbacks have not been resolved.

Noussairhamrit committed:
Update UserSettingSecurityWindow.vue

clean up code

rdenarie reviewed (changes requested):
2 feedbacks are still not resolved.

Noussairhamrit commented:
I don't understand if i will add the spaces or remove it in line 146 and 153.
because in both cases you told me not resolved.
or is it another error?

Noussairhamrit commented:
> 2 feedbacks are still not resolved.
